### ISSUE TITLE
Determine right value for maximum instance/total memory

### ISSUE THREAD

**Issue by raulk** [filecoin-project/ref-fvm#369]
## Context

For M1 (nv16), we are going with unlimited max pages given that the system can only run trusted logic via builtin system actors. However, it's not clear what the value of initial pages should be, in order to avoid excessive memory expansion operations.

## Request

- Run test vectors tracing memory expansions.
- Get a sense of average and maximum memory usage.
- Feed this info into #341.

Note: we no longer care about initial pages. The wasm module itself specifies this.

**Comment by aakoshh** [comment]
> Run test vectors tracing memory expansions.

@raulk are there existing test vectors you are using?

**Comment by Stebalien** [comment]
@aakoshh https://github.com/filecoin-project/ref-fvm/tree/release/v2/testing/conformance

**Comment by aakoshh** [comment]
Stats are in https://github.com/filecoin-project/ref-fvm/pull/1036

**Comment by Stebalien** [comment]
TL;DR:

- We currently use at most 11MiB for all the test vectors.
- We've set the current limit to 512MiB (both per instance, and for the entire stack).
- We haven't tested on mainnet yet, but we probably should. That means:
  - Taking the `mem-851-memory-expansion-gas` branch (which is based on release/v2).
  - Hacking something together to use the `TestLimiter` with a "normal" machine.
  - Printing stats to stdout.

However, we can adjust this post-testnet.

**Comment by Stebalien** [comment]
We should also consider the max stack height. Currently, our chosen memory limit is too small to hit the maximum stack height. We probably need to double it (at least) if we want to get there. For the recursive multisig test-case in lotus, we'd need 1.5GiB of maximum stack space.

On the other hand, do we really care?

**Comment by Stebalien** [comment]
I'm raising this to 2GiB for now so we can comfortably hit the maximum recursion limit.
